### PR TITLE
input: add a paw32xx driver

### PR DESCRIPTION
**Pull request by fabiobaltieri**
Hey, one more (probably last from me) low power mouse sensor driver. :-) I tested this on a board with what should be a PAW3222 (at least that's what the AliExpress seller claims), made with the pinout from a PAW3220 datasheet (the 3222 one does not even have that) and with the register description from a PAW3212 datasheet (neither of the other two has any register information, but the default register I have on my device seems to match this one, so either the basic registers are compatible or I don't know what I have).

Not totally sure what the intended CPI range should be on this since I don't have the right datasheet, but the scaling factor seems to match and the whole range seems to work, so this should be a decent starting point.

The code is pretty much a collage of the other two Pixart drivers, the chips have a lot in common.

--

Add a driver for PAW32xx sensors.

### CODE REVIEW

File: drivers/input/input_paw32xx.c [hunk 1]
```diff
@@ -0,0 +1,443 @@
...
+			work, struct paw32xx_data, motion_work);
+	const struct device *dev = data->dev;
+	const struct paw32xx_config *cfg = dev->config;
+	uint8_t val;
+	int16_t x, y;
+	int ret;
+
+	ret = paw32xx_read_reg(dev, 2, &val);
+	if (ret < 0) {
+		return;
```
Comment (faxe1008) [inline drivers/input/input_paw32xx.c:166]: Wondering if some added logging would be a good idea here - it makes the code noisier (you could move it into the helper methods)

File: drivers/input/input_paw32xx.c [hunk 2]
```diff
@@ -0,0 +1,443 @@
...
+{
+	uint8_t val;
+	int ret;
+
+	ret = paw32xx_read_reg(dev, addr, &val);
+	if (ret < 0) {
+		return ret;
+	}
+
+	val = (val & ~mask) | (value & mask);
```
Comment (faxe1008) [inline drivers/input/input_paw32xx.c:144]: Maybe we could use a util macro for this at some point :^)

File: drivers/input/input_paw32xx.c [hunk 3]
```diff
@@ -0,0 +1,443 @@
...
+		val = 0;
+		break;
+	default:
+		return -ENOTSUP;
+	}
+
+	ret = paw32xx_update_reg(dev, PAW32XX_CONFIGURATION,
+				 CONFIGURATION_PD_ENH, val);
+	if (ret < 0) {
+		return ret;
```
Comment (faxe1008) [inline drivers/input/input_paw32xx.c:409]: Nit: just return ret here, if statement not needed

Comment (fabiobaltieri) [reply drivers/input/input_paw32xx.c:409]: not a fan of that pattern to be totally honest, I always find myself hacking something in the code and having to add the explicit return so I can, for example, print a value after the function

File: drivers/input/input_paw32xx.c (hunk 2 above)
Comment (fabiobaltieri) [reply drivers/input/input_paw32xx.c:144]: may be an interesting followup, the i2c helpers are very useful, sure spi could have some

File: drivers/input/input_paw32xx.c (hunk 1 above)
Comment (fabiobaltieri) [reply drivers/input/input_paw32xx.c:166]: not much into it to be honest, the device is rather simple

I did forget to use the definition here though (that "2"), fixed

**Review by faxe1008 — DISMISSED** [review]

**Comment by fabiobaltieri** [comment]
@faxe1008 push another update, was skimming through the datasheet and realized that there's a way of reading multiple registers in a single transaction, just updated the read function to do that for the xy data

**Review by faxe1008 — APPROVED** [review]

**Review by aasinclair — APPROVED** [review]

**Review by kartben — APPROVED** [review]

**Pull request merged**